runtime/dom-context.js

```diff
@@ -101,7 +105,7 @@ class DomContext {
     // TODO(sjmiles): remember that attribute names from HTML are lower-case
     return node[name] || node.getAttribute(name);
   }
-  initInnerContexts(slotSpec) {
+  initInnerContexts(slotSpec, globalSubId) {
```

Comment:
IMO this method is too complex, suggest factoring it.

Reply:
done